shell/common/rasterizer.h

```diff
@@ -374,8 +374,29 @@ class Rasterizer final {
   ///
   /// @param[in]  max_bytes  The maximum byte size of resource that may be
   ///                        cached for GPU rendering.
+  /// @param[in]  from_user  Whether this request was from user code, e.g. via
+  ///                        the flutter/skia message channel, in which case
+  ///                        it should not be overridden by the platform.
   ///
-  void SetResourceCacheMaxBytes(int max_bytes);
+  void SetResourceCacheMaxBytes(size_t max_bytes, bool from_user);
+
+  //----------------------------------------------------------------------------
+  /// @brief      The current value of Skia's resource cache size, if a surface
+  ///             is present.
+  ///
+  /// @attention  This cache setting will be invalidated when the surface is
+  ///             torn down via `Rasterizer::Teardown`. This call must be made
+  ///             again with new limits after surface re-acquisition.
+  ///
+  /// @attention  This cache does not describe the entirety of GPU resources
+  ///             that may be cached. The `RasterCache` also holds very large
+  ///             GPU resources.
+  ///
+  /// @see        `RasterCache`
+  ///
+  /// @return     The size of Skia's resource cache, if available.
+  ///
+  std::optional<size_t> GetResourceCacheMaxBytes() const;
```

Comment:
Per discussion above, this can now be just `size_t GetResourceCacheMaxBytes() const;`